### PR TITLE
Refactor code v2

### PR DESCRIPTION
**Pull request by bennychanya**
Refactor boolean conditions for clarity
Edited Javadoc for MarkAsDoneCommand
Edited checkModuleExist method to allow module to be added again to SemModuleList if the existing module received a F or CU grade

### CODE REVIEW

**Comment by RenzoTsai** [comment]
It seems there is bug in `add id/ACC1004 s/5 mc/4` in text ui test. This module can be added to semester modules list though it is already in the list.

**Comment by RenzoTsai** [comment]
Looks good!  But there is an issue, when user re-add a failed module to semester, there are two modules which have the same id in semester module list. So when they try to delete the module, they may face a problem. So is it necessary to keep this function or just ask them delete failed modules first and then re-add the module.
image

**Comment by RenzoTsai** [comment]
> Looks good! But there is an issue, when user re-add a failed module to semester, there are two modules which have the same id in semester module list. So when they try to delete the module, they may face a problem. So is it necessary to keep this function or just ask them delete failed modules first and then re-add the module.
> image

To solve the problem, I made a [PR](https://github.com/bennychanya/tp/pull/3) to your branch. Please check it, thanks!

**Comment by bennychanya** [comment]
Hi Runze, i just saw your PR. Failed modules do count to your grades and CAP. Is there a way to delete it without affecting our CAP?

**Comment by RenzoTsai** [comment]
> Hi Runze, i just saw your PR. Failed modules do count to your grades and CAP. Is there a way to delete it without affecting our CAP?

Oh, you mean if users study a failed module again and get A, the grade of this module will not be changed from F to A directly, but will keeps both F and A?

**Comment by bennychanya** [comment]
That is correct.

**Comment by bennychanya** [comment]
> > Hi Runze, i just saw your PR. Failed modules do count to your grades and CAP. Is there a way to delete it without affecting our CAP?
> 
> Oh, you mean if users study a failed module again and get A, the grade of this module will not be changed from F to A directly, but will keeps both F and A?

That is correct. For F grades, it is counted into CAP and cannot be changed

File: src/main/java/seedu/duke/command/AddToSemCommand.java [hunk 1]
```diff
@@ -79,17 +79,22 @@ private void checkAvailableModulesList(SelectedModule selectedModule) {
...
         for (SemModulesList sem: semesterList) {
-            if (sem.isInList(selectedModule.getName()) || sem.isInList(selectedModule.getId())) {
-                return true;
+            for (SelectedModule module: sem) {
+                boolean hasSameModuleId = module.getId().equals(selectedModule.getId()) && module.isIdValid();
+                boolean hasSameModuleName = module.getName().equals(selectedModule.getName()) && module.isNameValid();
+                if (hasSameModuleId || hasSameModuleName) {
+                    return true;
+                    }
+                }
```
Comment (RenzoTsai) [inline src/main/java/seedu/duke/command/AddToSemCommand.java]: These two parts(deleted and new added) seem to have the same function or meaning haha.

File: src/main/java/seedu/duke/command/AddToSemCommand.java [hunk 2]
```diff
@@ -79,15 +79,20 @@ private void checkAvailableModulesList(SelectedModule selectedModule) {
...
+     * @param semesterList user's semester list
      * @return boolean value of true if the module is in the user's semester list, and false otherwise
      */
     private boolean checkModuleExist(SemesterList semesterList) {
         for (SemModulesList sem: semesterList) {
-            if (sem.isInList(selectedModule.getName()) || sem.isInList(selectedModule.getId())) {
-                return true;
+            for (SelectedModule module : sem) {
+                boolean hasSameModuleId = module.getId().equals(selectedModule.getId()) && module.isIdValid();
+                boolean hasSameModuleName = module.getName().equals(selectedModule.getName()) && module.isNameValid();
```
Comment (DeetoMok) [inline src/main/java/seedu/duke/command/AddToSemCommand.java:92]: Can we use equalsignorecase instead?

File: src/main/java/seedu/duke/command/MarkAsDoneCommand.java [hunk 3]
```diff
@@ -28,17 +28,26 @@ public MarkAsDoneCommand(String description, Grading grade) {
...
+     * @param semesterList Semester List containing Semester Module Lists, which contains selected modules
+     * @throws RuntimeException throws except if module not found in semester
+     */
     private void markAsDoneCommand(SemesterList semesterList) throws RuntimeException {
         for (SemModulesList sem: semesterList) {
             for (SelectedModule module: sem) {
-                if (module.getName().equals(description) || module.getId().equals(description)) {
-                    if (grade != Grading.F && grade != Grading.CU && !module.getDone()) {
+                boolean isModuleName = module.getName().equals(description);
+                boolean isModuleId = module.getId().equals(description);
```
Comment (DeetoMok) [inline src/main/java/seedu/duke/command/MarkAsDoneCommand.java]: same for here, can we use equalsignorecase?

File: src/main/java/seedu/duke/command/AddToSemCommand.java (hunk 2 above)
Comment (bennychanya) [reply src/main/java/seedu/duke/command/AddToSemCommand.java:92]: I think we can use equalsignorecase

File: src/main/java/seedu/duke/command/MarkAsDoneCommand.java (hunk 3 above)
Comment (bennychanya) [reply src/main/java/seedu/duke/command/MarkAsDoneCommand.java]: Same here

**Pull request merged**